### PR TITLE
Typescript 5.6

### PR DESCRIPTION
**Pull request by turadg**
_evergreen_

## Description

https://devblogs.microsoft.com/typescript/announcing-typescript-5-6

Notable changes for this repo,
- [Strict Builtin Iterator Checks (and --strictBuiltinIteratorReturn)](https://devblogs.microsoft.com/typescript/announcing-typescript-5-6-beta/#strict-builtin-iterator-checks-and---strictbuiltiniteratorreturn)

### Security Considerations
n/a

### Scaling Considerations
n/a

### Documentation Considerations
none

### Testing Considerations
I captured type coverage before and after.

### Upgrade Considerations
n/a

### CODE REVIEW

File: packages/internal/src/utils.js [hunk 1]
```diff
@@ -269,6 +269,10 @@ export const synchronizedTee = (sourceStream, readerCount) => {
       [Symbol.asyncIterator]() {
         return reader;
       },
+      // eslint-disable-next-line no-restricted-globals
+      [Symbol.asyncDispose]() {
+        throw Error('asyncDispose not implemented.');
+      },
```
Comment (mhofman) [inline packages/internal/src/utils.js]: According to the [proposal](https://arai-a.github.io/ecma262-compare/?pr=3000&id=sec-%25asynciteratorprototype%25-%40%40asyncdispose), on async iterators this is the equivalent of calling `return()`.

```suggestion
      async [Symbol.asyncDispose]() {
        await reader.return();
      },
```

**Review by mhofman — COMMENTED** [review]
My understanding is that 5.6 went final last night, so let's switch to that instead of the rc. It also seems that they may have reverted some changes that caused some of the Node and Iterator types compatibility issues that had to be worked around here, so let's possibly consider stashing those for now.

**Comment by mhofman** [comment]
> @turadg [force-pushed](https://github.com/Agoric/agoric-sdk/compare/e119569b394e27cdee6718dfafddc04ba90cfaed..7da790219da751f6f80bcdd75811558a0e3d621e) the typescript-5.6 branch from [e119569 ](https://github.com/Agoric/agoric-sdk/commit/e119569b394e27cdee6718dfafddc04ba90cfaed)to [7da7902 ](https://github.com/Agoric/agoric-sdk/commit/7da790219da751f6f80bcdd75811558a0e3d621e) [1 hour ago](https://github.com/Agoric/agoric-sdk/pull/9795#event-14218781319)

As a matter of process, it's really difficult to review what changed from my previous review when the base commit this PR is staged on changed since the comparison of changes since last review is now broken. In general while the review is in progress, prefer merging changes from master if you must update the PR so that changes remain additive.

> It also seems that they may have reverted some changes that caused some of the Node and Iterator types compatibility issues that had to be worked around here

Have we tried reverting the version bump of `@types/node` ? I am somewhat concerned about adding typing information for an API level we do not support.

**Comment by turadg** [comment]
> As a matter of process…

I'll leave process discussions to internal forums.

> Have we tried reverting the version bump of `@types/node` ? I am somewhat concerned about adding typing information for an API level we do not support.

Reverting the commit cause dozens of errors, all with the Buffer type due to IterableIterator types:
```
packages/internal/src/netstring.js:17:25 - error TS2322: Type 'Buffer' is not assignable to type 'Uint8Array'.
  The types returned by 'entries()' are incompatible between these types.
    Type 'IterableIterator<[number, number]>' is missing the following properties from type 'ArrayIterator<[number, number]>': map, filter, take, drop, and 9 more.

17   return Buffer.concat([prefix, data, suffix]);
                           ~~~~~~
```

We run CI tests with Node 18 so the only risk of Node 22 types is incorrect static analysis.

**Comment by mhofman** [comment]
> due to IterableIterator types

Ah got it, it's due to Iterator helpers addition, not the other iterable object type change.

The problem is most likely because we target `esnext` in our tsconfig, instead of being more conservative.

> We run CI tests with Node 18 so the only risk of Node 22 types is incorrect static analysis.

Of course, my goal was to keep the tools working as best as possible, and the development iteration short.

**Review by mhofman — APPROVED** [review]

**Comment by turadg** [comment]
Bypassed integration because this PR has passed integration several times save for multichain-testing, and I just reverted the changes to that.

**Pull request merged**